lib/viewcomponents/node.jsx

```diff
@@ -1,23 +1,68 @@
 var React = require('react');
+var IO = require('./io.jsx');
 var mouseDownDrag = require('../utilities/mouseDownDrag');
+var PositionUtils = require('../utilities/positionUtils.js');
+var CardinalDirection = require('../enum/cardinalDirection.js');
+var IOLabelPosition = require('../enum/ioLabelPosition');
+
+var nodeWidth = 150;
+var nodeHeight = nodeWidth/1.6;
+var headerHeight = 35;
+var padding = 5;
+
+var ioRadius = 4;
+var ioSpacing = 15;
+
+// DATA MODEL scales values 0-100. Undo that.
+var DATA_MODEL_MULTIPLIER = 100.0;
+
+// TODO: specific to rectangular nodes. Refactor out.
+var DIRECTION_TO_LABEL_POSITION = {};
+
+DIRECTION_TO_LABEL_POSITION[CardinalDirection.NORTH] = IOLabelPosition.BELOW;
+DIRECTION_TO_LABEL_POSITION[CardinalDirection.SOUTH] = IOLabelPosition.ABOVE;
+DIRECTION_TO_LABEL_POSITION[CardinalDirection.EAST] = IOLabelPosition.LEFT;
+DIRECTION_TO_LABEL_POSITION[CardinalDirection.WEST] = IOLabelPosition.RIGHT;
+
 
 var Node = React.createClass({
 
     render: function() {
         return (
-            <g className='node'>
+            <g
+             className='node'
+             transform={'translate(' + this.props.x + ', ' + this.props.y + ')'} >
                 <rect
+                    height={nodeHeight} width={nodeWidth}
                     className='node-box'
-                    x={this.props.x} y={this.props.y} height='93.75' width='150'
                     onMouseDown={mouseDownDrag.bind(this, 'node_body', null, null, this._onNodeBodyPseudoDrag)} />
-                <text className='label' text-anchor='start' x={this.props.x + 5} y={this.props.y + 15}>
+                <text className='label' text-anchor='start' x={padding} y={padding + 10}>
                     {this.props.label}
                 </text>
-                {/* inputs and outputs */}
+                {this._getRenderedIOs(this.props.inputs, true)}
+                {this._getRenderedIOs(this.props.outputs, false)}
             </g>
         );
     },
 
+    _getRenderedIOs: function(ios, isInput) {
```

Comment:
It took me a while to figure out that these are models. It might make it more clear if the naming of these models were changed throughout this file to have names like ioModels, ioModel, nodeModel. This might be something to do later.